pkg/services/ngalert/store/loki_range_to_instant_test.go

```diff
@@ -0,0 +1,189 @@
+package store
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCanBeInstant(t *testing.T) {
+	tcs := []struct {
+		name     string
+		expected bool
+		rule     *models.AlertRule
+	}{
+		{
+			name:     "valid rule that can be migrated from range to instant",
+			expected: true,
+			rule:     createMigrateableLokiRule(t),
+		},
+		{
+			name:     "invalid rule where the data array is too short to be migrateable",
+			expected: false,
+			rule: func() *models.AlertRule {
+				r := createMigrateableLokiRule(t)
+				r.Data = []models.AlertQuery{r.Data[0]}
+				return r
+			}(),
+		},
+		{
+			name:     "invalid rule that is not a range query",
+			expected: false,
+			rule: func() *models.AlertRule {
```

Comment:
Massive nit, but since all of these start with `createMigrateableLokiRule` you could treat this like a mutator, have it be `func(r *models.AlertRule) {}` and have it just modify the rule in place. Then none of them need to construct the rule inline and you could drop the `return r` everywhere.

This is a super minor style thing though so feel free to ignore.